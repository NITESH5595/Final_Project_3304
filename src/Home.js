import React, { useState, useContext, useEffect } from "react";
import { auth, db } from "./base";
import { signOut } from "firebase/auth";
import { AuthContext } from "./AuthProvider";
// import "./Home.css";
import { useNavigate } from "react-router-dom";
import { ref, onValue } from "firebase/database";
import "./index.css";

function Home() {
  const { currentUser } = useContext(AuthContext);
  const [username, setUsername] = useState("");
  const navigate = useNavigate();
  const [time, setTime] = useState(Date.now());

  useEffect(() => {
    const interval = setInterval(() => setTime(Date.now()), 1000);
    return () => {
      clearInterval(interval);
    };
  }, []);

  useEffect(() => {
    if (currentUser) {
      const starCountRef = ref(db, "users/" + currentUser.uid);
      onValue(starCountRef, (snapshot) => {
        if (snapshot.exists()) {
          var data = snapshot.val();
          setUsername(data.firstName + " " + data.lastName);
        }
      });
    }
  }, [currentUser]);

  const clickLogin = () => {
    if (currentUser) {
      signOut(auth);
    } else {
      navigate("/login");
    }
  };

  const clickSignup = () => {
    navigate("/signup");
  };
  const aboutus = () => {
    navigate("/about");
  };
  const services = () => {
    navigate("/services");
  };
  const customers = () => {
    navigate("/customers");
  };
  const weather = () => {
    navigate("/weather");
  };

  var myCurrentDate = new Date();
  var date =
    myCurrentDate.getFullYear() +
    "-" +
    (myCurrentDate.getMonth() + 1) +
    "-" +
    myCurrentDate.getDate() +
    " " +
    myCurrentDate.getHours() +
    ":" +
    myCurrentDate.getMinutes() +
    ":" +
    myCurrentDate.getSeconds();
  const newCurrentDate = date;

  return (
    <div className="mainContainer">
      {currentUser && (
        <div>
          <div className="navbar">
            <ul className="menu">
              <li className="item">
                <button className="menuBtn" onClick={aboutus}>
                  About Us
                </button>
                {/* <a href={<AboutUs />}>About Us</a> */}
              </li>
              <li className="item">
                <button className="menuBtn" onClick={services}>
                  Our Services
                </button>
              </li>
              <li className="item">
                <button className="menuBtn" onClick={customers}>
                  Our Customers
                </button>
              </li>
            </ul>
          </div>
          <div className="main">
            <h1>Welcome {username}</h1>
            <h3>Time and Date</h3>

            <div className="date">
              Current Date and Time
              <br />
              {newCurrentDate}
            </div>
            <div>
              <button onClick={weather} className="btn">
                Weather
              </button>
            </div>
          </div>
        </div>
      )}
      <div className="buttons">
        <button onClick={clickLogin} className="btn">
          {currentUser ? "Log Out" : "Login"}
        </button>
        {!currentUser && (
          <button onClick={clickSignup} className="btn">
            Sign Up
          </button>
        )}
      </div>
    </div>
  );
}

export default Home;
