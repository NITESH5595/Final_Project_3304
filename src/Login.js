import React, { useState } from "react";
import { signInWithEmailAndPassword } from "firebase/auth";
import "./index.css";
import { useNavigate } from "react-router-dom";
import { auth } from "./base";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const navigate = useNavigate();

  const handleSubmit = (e) => {
    e.preventDefault();
    function onRegister() {
      signInWithEmailAndPassword(auth, email, password).catch((error) =>
        console.log(error)
      );
      navigate("/");
    }
    onRegister();
  };

  const goBack = () => {
    navigate("/");
  };

  return (
    <div className="mainContainer">
      <form className="loginForm" onSubmit={handleSubmit}>
        <input
          placeholder="email"
          type="email"
          onChange={(e) => setEmail(e.target.value)}
        ></input>
        <input
          placeholder="password"
          type="password"
          onChange={(e) => setPassword(e.target.value)}
        ></input>
        <button className="btn">Login</button>
        <button className="btn" onClick={() => goBack()}>
          Go Back
        </button>
      </form>
    </div>
  );
};

export default Login;
