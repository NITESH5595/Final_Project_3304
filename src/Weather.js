import React, { useState } from "react";
import "./index.css";
import { useNavigate } from "react-router-dom";
import axios from "axios";

function Weather() {
  const [data, setData] = useState({});
  const [location, setLocation] = useState("");

  const url = `https://api.openweathermap.org/data/2.5/weather?q=${location}&units=metric&appid=2f289d6f6e94bd9e23ed6991c58fe88c`;

  const searchLocation = (event) => {
    if (event.key === "Enter") {
      axios.get(url).then((response) => {
        setData(response.data);
        console.log(response.data);
      });
      setLocation("");
    }
  };

  const navigate = useNavigate();
  const goBack = () => {
    navigate("/");
  };

  return (
    <div className="weatherApp">
      <div className="weatherContainer">
        <div className="search">
          <input
            value={location}
            onChange={(event) => setLocation(event.target.value)}
            placeholder="Enter Location"
            type="text "
            onKeyPress={searchLocation}
          />
        </div>
        <div className="top">
          <div className="location">
            <h1>{data.name}</h1>
          </div>
          <div className="temp">
            {data.main ? <h1>{data.main.temp.toFixed()}&deg;C</h1> : null}
          </div>
          <div className="desc">
            {data.weather ? <p>{data.weather[0].main}</p> : null}
          </div>
        </div>
        {data.name != undefined && (
          <div className="bottom">
            <div className="feels">
              {data.main ? (
                <p className="bold">{data.main.feels_like.toFixed()}&deg;C</p>
              ) : null}
              <p>Feels Like</p>
            </div>
            <div className="humidity">
              {data.main ? <p className="bold">{data.main.humidity}%</p> : null}
              <p>Humidity</p>
            </div>
            <div className="wind">
              {data.wind ? (
                <p className="bold">{data.wind.speed.toFixed()}</p>
              ) : null}
              <p>Wind Speed</p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}

export default Weather;
