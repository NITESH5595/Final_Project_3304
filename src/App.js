import React from "react";
import { BrowserRouter, Routes, Route, Router } from "react-router-dom";
import Home from "./Home";
import Login from "./Login";
import SignUp from "./Signup";
import { AuthProvider } from "./AuthProvider";
import About from "./AboutUs";
import Services from "./Services";
import Customers from "./Customers";
import Weather from "./Weather";

function App() {
  return (
    <AuthProvider>
      <div className="app">
        <BrowserRouter>
          <Routes>
            <Route exact path="/" element={<Home />} />
            <Route exact path="/about" element={<About />} />
            <Route exact path="/services" element={<Services />} />
            <Route exact path="/customers" element={<Customers />} />
            <Route exact path="/weather" element={<Weather />} />
            <Route exact path="/login" element={<Login />} />
            <Route exact path="/signup" element={<SignUp />} />
          </Routes>
        </BrowserRouter>
      </div>
    </AuthProvider>
  );
}

export default App;
// npm run test -- --coverage --watchAll=false; for code coverage
