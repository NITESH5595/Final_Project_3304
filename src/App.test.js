import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import App from "./App";
import Home from "./Home";
import Weather from "./Weather";
import Login from "./Login";

describe("App", () => {
  it("renders without crashing given the required props", () => {
    const props = {
      isFetching: false,
      dispatch: jest.fn(),
      selectedSubreddit: "reactjs",
      posts: [],
    };
    const wrapper = shallow(<Login />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
