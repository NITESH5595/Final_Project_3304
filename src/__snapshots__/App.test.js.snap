// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<AuthProvider>
  <div
    className="app"
  >
    <BrowserRouter>
      <Routes>
        <Route
          element={<Home />}
          exact={true}
          path="/"
        />
        <Route
          element={<About />}
          exact={true}
          path="/about"
        />
        <Route
          element={<Services />}
          exact={true}
          path="/services"
        />
        <Route
          element={<Customers />}
          exact={true}
          path="/customers"
        />
        <Route
          element={<Weather />}
          exact={true}
          path="/weather"
        />
        <Route
          element={<Login />}
          exact={true}
          path="/login"
        />
        <Route
          element={<SignUp />}
          exact={true}
          path="/signup"
        />
      </Routes>
    </BrowserRouter>
  </div>
</AuthProvider>
`;
